File: 3621-minimum-operations-to-make-array-values-equal-to-k/minimum-operations-to-make-array-values-equal-to-k.py
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        if any(num < k for num in nums):
            return -1
        nums.sort(reverse=True)
        operations = 0
        current_max = nums[0]

        for num in nums:
            if num > k and num < current_max:
                current_max = num
                operations += 1
        if current_max > k:
            operations += 1
        return operations

File: 3621-minimum-operations-to-make-array-values-equal-to-k/minimum-operations-to-make-array-values-equal-to-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var minOperations = function(nums, k) {
    const uniqueNums = [... new Set(nums)];
    let result = 0;
    for (const num of uniqueNums) {
        if (num < k) {
            return -1;
        } else if (num > k) {
            result++;
        }
    }
    return result;
};